source/actors/player/Item.hx

```diff
@@ -0,0 +1,80 @@
+package actors.player;
+
+import flixel.util.FlxTimer;
+import flixel.FlxG;
+import flixel.FlxSprite;
+
+class Item extends FlxSprite 
+{
+    public var sWidth(default, never):Int = 50;
+    public var sHeight(default, never):Int = 10;
+
+    // 0 = right, 1 = down, 2 = left, 3 = up
+    private var lastUsed:Int = 0;
+
+    private var time:FlxTimer;
+    private var delay:Float = 0.2;
+
+    public function new() {
+        super();
+    }
+    override public function update(elapsed:Float):Void
+        {
+            if(FlxG.keys.pressed.RIGHT)
+                {
+                    lastUsed = 0;
+                }
+            else if(FlxG.keys.pressed.DOWN)
+                {
+                    lastUsed = 1;
+                }
+            else if(FlxG.keys.pressed.LEFT)
+                {
+                    lastUsed = 2;
+                }
+            else if(FlxG.keys.pressed.UP)
+                {
+                    lastUsed = 3;
+                }
+            if(time.finished)
+            {
+                time.cancel();
+                //kill();
+            }
```

Comment:
I don't think this `cancel` call should be needed, and it might be messing with the timer's execution of the `OnComplete` function that you set up? (`stopSwing`, in this case)

I'd need to do some direct testing to know for sure.